File: Problems/2870. Minimum Number of Operations to Make Array Empty/2870. Minimum Number of Operations to Make Array Empty.py
from collections import defaultdict
from typing import List
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        count = defaultdict(int)
        res = 0
        
        for num in nums:
            count[num] += 1
        for num in count:
            if count[num] == 1: return -1
            res += (count[num]+2) // 3
        
        return res

File: Problems/2870. Minimum Number of Operations to Make Array Empty/2870. Minimum Number of Operations to Make Array Empty.java
import java.util.HashMap;
class Solution {
    public int minOperations(int[] nums) {
        Arrays.sort(nums);
        int res = 0;
        int count = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i-1]) count++;
            else {
                if (count == 1) return -1;
                res += (count+2) / 3;
                count = 1;
            }
        }
        if (count == 1) return -1;
        res += (count+2) / 3;

        return res;
    }
}